admin-auth/src/test/kotlin/com/epam/drill/admin/auth/UserRepositoryImplTest.kt

```diff
@@ -15,19 +15,234 @@
+            assertEquals("hash2", it[UserTable.passwordHash])
+            assertEquals("ADMIN", it[UserTable.role])
+            assertTrue(it[UserTable.deleted])
+            assertTrue(it[UserTable.blocked])
+        }
     }
 
     @Test
-    fun `given`() {
+    fun `findAllNotDeleted must return all not deleted users`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..15) {
+            it[deleted] = true //insert 5 deleted users
+        }
+        insertRandomUsers(16..20)
+
+        val users = repository.findAllNotDeleted()
+
+        assertEquals(20 - 5, users.size)
+    }
+
+    @Test
+    fun `given existing username, findByUsername must return user`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..11) {
+            it[username] = "foobar"
+        }
+        insertRandomUsers(12..20)
+
+        val user = repository.findByUsername("foobar")
+
+        assertNotNull(user)
+        assertEquals("foobar", user.username)
+    }
+
+    @Test
+    fun `findByUsername must return even blocked user`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..11) {
+            it[username] = "foobar"
+            it[blocked] = true
+        }
+        insertRandomUsers(12..20)
+
+        val user = repository.findByUsername("foobar")
 
+        assertNotNull(user)
+        assertEquals("foobar", user.username)
+        assertTrue(user.blocked)
     }
 
+    @Test
+    fun `findByUsername mustn't return deleted user`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..11) {
+            it[username] = "foobar"
+            it[deleted] = true
+        }
+        insertRandomUsers(12..20)
+
+        val user = repository.findByUsername("foobar")
+
+        assertNull(user)
+    }
+
+    @Test
+    fun `given case insensitive username, findByUsername must return user`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..11) {
+            it[username] = "FooBar"
+        }
+
+        val user = repository.findByUsername("foobar")
+
+        assertNotNull(user)
+        assertEquals("FooBar", user.username)
+    }
+
+    @Test
+    fun `given existing id, findById must return user`() = withTransaction {
+        val id = insertRandomUsers(1..10).random()
+
+        val user = repository.findById(id)
+
+        assertNotNull(user)
+        assertEquals(id, user.id)
+    }
+
+    @Test
+    fun `given non-existent id, findById must return null`() = withTransaction {
+        val ids = insertRandomUsers(1..10)
+        assertFalse { ids.contains(404) }
+
+        val user = repository.findById(404)
+
+        assertNull(user)
+    }
+
+    @Test
+    fun `findById must return even deleted user`() = withTransaction {
+        insertRandomUsers(1..10)
+        val id = insertRandomUsers(11..15) {
+            it[deleted] = true
+        }.random()
```

Comment:
todo - remove random and either
- write a single case with a single id 
- craft specific cases to treat edge conditions
- iterate over all entries